### PR TITLE
enable vpc-flow-logs

### PR DESCRIPTION
**Pull request by chandra0608**
## Description

Please provide a brief description of the changes introduced by this pull request.

## Related Issue

- [Issue Number](link-to-issue): Brief description

## Checklist

Please ensure that the following steps are completed before submitting the pull request:

- [x] Code follows the Terraform best practices and style guidelines.
- [x] Changes are appropriately documented, including any necessary updates to README or other documentation files.
- [x] Unit tests have been added or updated to cover the changes introduced by this pull request.
- [x] Changes have been tested locally and verified to work as expected.
- [x] The code has been reviewed to ensure it aligns with the project's goals and standards.
- [x] Dependencies and backward compatibility have been considered and addressed if applicable.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Intermediate change (work in progress)

## Testing Instructions

Provide detailed steps or instructions for testing the changes introduced by this pull request.

## Screenshots

Include any relevant screenshots or visual aids to help reviewers understand the changes visually, if applicable.

## Additional Notes

Add any additional notes or context that might be helpful for reviewers or users testing the changes.

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -172,3 +172,54 @@ variable "tags" {
...
+  default     = true
+  description = "Specifies whether key rotation is enabled"
+}
+
+variable "enabled" {
+  type        = bool
+  default     = true
+  description = "Set to false to prevent the module from creating any resources"
+}
+
```
Comment (vijay-stephen) [inline variables.tf:199]: this variable is not required

File: variables.tf [hunk 2]
```diff
@@ -172,3 +172,54 @@ variable "tags" {
...
+  type        = number
+  default     = 10
+  description = "Duration in days after which the key is deleted after destruction of the resource"
+}
+
+variable "enable_key_rotation" {
+  type        = bool
+  default     = true
+  description = "Specifies whether key rotation is enabled"
+}
```
Comment (vijay-stephen) [inline variables.tf:192]: KMS related variable has to be one variable using object

File: variables.tf [hunk 3]
```diff
@@ -172,3 +172,54 @@ variable "tags" {
   description = "(optional) Tags for VPC resources"
   default     = {}
 }
+
+variable "enable_vpc_flow_log_to_cloudwatch" {
+  type        = bool
+  description = "Flag to enable or disable VPC flow logs to Cloudwatch."
+  default     = false
+}
```
Comment (vijay-stephen) [inline variables.tf:180]: Pls rename it as `enable_vpc_flow_log`

File: variables.tf [hunk 4]
```diff
@@ -172,3 +172,54 @@ variable "tags" {
...
+
+variable "acl" {
+  type        = string
+  default     = "private"
+  description = <<-EOT
+    Please node ACL is deprecated by AWS in favor of bucket policies.
+    Defaults to "private" for backwards compatibility,recommended to set `s3_object_ownership` to "BucketOwnerEnforced" instead.
+  EOT
+}
```
Comment (vijay-stephen) [inline variables.tf:225]: this variable has to be removed, S3 bucket has to be created by root module

File: locals.tf [hunk 5]
```diff
@@ -12,5 +13,75 @@ locals {
...
+          object_size_less_than    = "1000000"
+          prefix                   = "logs/"
+          tags = {
+            "environment" = "production"
+            "department"  = "IT"
+          }
+        }
+      }
+    ]
+  }
```
Comment (vijay-stephen) [inline locals.tf:86]: S3 bucket has to be created from root module

File: main.tf [hunk 6]
```diff
@@ -145,3 +145,114 @@ resource "aws_route_table_association" "additional" {
   subnet_id      = aws_subnet.this[each.value.key].id
   route_table_id = aws_route_table.this[each.value.key].id
 }
+
+# Module for KMS Key Management
+module "kms" {
+  source                  = "sourcefuse/arc-kms/aws"
+  version                 = "1.0.9"
+  count                   = var.enable_vpc_flow_log_to_cloudwatch ? 1 : 0
+  enabled                 = var.enabled
```
Comment (vijay-stephen) [inline main.tf:154]: remove enable variable

File: main.tf [hunk 7]
```diff
@@ -145,3 +145,114 @@ resource "aws_route_table_association" "additional" {
...
+  acl              = var.acl
+  lifecycle_config = local.lifecycle_config
+  tags = merge(
+    {
+      Name = "${var.name}-vpc-flowlogs"
+    },
+    var.tags
+  )
+}
+
```
Comment (vijay-stephen) [inline main.tf:182]: remove it, it has be to be created from root module

**Review by vijay-stephen — CHANGES_REQUESTED** [review]

File: variables.tf (hunk 1 above)
Comment (chandra0608) [reply variables.tf]: corrections made

File: variables.tf (hunk 2 above)
Comment (chandra0608) [reply variables.tf]: corrections made

File: variables.tf (hunk 3 above)
Comment (chandra0608) [reply variables.tf]: corrections made

File: variables.tf (hunk 4 above)
Comment (chandra0608) [reply variables.tf]: corrections made

File: main.tf (hunk 6 above)
Comment (chandra0608) [reply main.tf]: coorrections made

File: locals.tf (hunk 5 above)
Comment (chandra0608) [reply locals.tf]: corrections made

**Review by vijay-stephen — APPROVED** [review]

**Pull request merged**